lib/test/cdo/test_high_frequency_reporter.rb

```diff
@@ -0,0 +1,76 @@
+# require_relative '../test_helper'
+# Temporary:
+require 'minitest/autorun'
+require 'cdo/high_frequency_reporter'
+require 'tempfile'
+require 'timecop'
+
+class HighFrequencyReporterTest < MiniTest::Test
+  # Second time an event happens, report it to slack
+  def test_report_on_second_event
+    # The first time the error occurs, nothing should be reported
+    # to Slack.  The Mock with no expectations ensures that for us.
+    fake_slack = MiniTest::Mock.new # expects nothing
+
+    Tempfile.create do |f|
+      reporter = HighFrequencyReporter.new(fake_slack, 'fake_channel', f.path)
+      reporter.load
+      reporter.record "Something happened"
+      # Mock will fail here if it is called
+      reporter.report!
+      reporter.save
+
+      # The second time the same error occurs, it should be reported
+      # to Slack.  We add a mock expectation to check that.
+      fake_slack.expect :message, nil, [String, Hash]
```

Comment:
TIL `exepct` use `===` to check argument types (String and Hash in this case).
https://www.rubydoc.info/gems/minitest/5.11.3/Minitest%2FMock:expect

Reply:
Whoa... === does not do what I would expect it to in Ruby.